cmd/icingadb/main.go

```diff
@@ -285,9 +289,43 @@ func checkDbSchema(ctx context.Context, db *icingadb.DB) error {
 		return errors.Wrap(err, "can't check database schema version")
 	}
 
-	if version != expectedSchemaVersion {
-		return errors.Errorf("expected database schema v%d, got v%d", expectedSchemaVersion, version)
+	if version != expectedDbSchemaVersion {
+		return errors.Errorf("expected database schema v%d, got v%d", expectedDbSchemaVersion, version)
 	}
 
 	return nil
 }
+
+// monitorRedisSchema monitors rc's icinga:schema version validity.
+func monitorRedisSchema(logger *zap.SugaredLogger, rc *icingaredis.Client, pos string) {
+	for {
+		pos = checkRedisSchema(logger, rc, pos)
+	}
+}
+
+// checkRedisSchema verifies rc's icinga:schema version.
+func checkRedisSchema(logger *zap.SugaredLogger, rc *icingaredis.Client, pos string) (newPos string) {
+	var logWait func(...interface{})
+	if pos == "0-0" {
+		logWait = logger.Info
+	} else {
+		logWait = logger.Debug
+	}
+
+	logWait("Waiting for Redis schema version")
+
+	cmd := rc.XRead(context.Background(), &redis.XReadArgs{Streams: []string{"icinga:schema", pos}})
```

Comment:
I'm a fan of "no news is good news". I suggest handling the first `XREAD` differently: typically, a non-blocking read should return the schema version just fine, then there's no need to log anything at level info and only if nothing is returned, log the waiting message and do a blocking read.